src/overlays/actors/ovl_En_Mkk/z_en_mkk.c

```diff
@@ -82,62 +111,498 @@ static DamageTable D_80A4F754 = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mkk/func_80A4EDF0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mkk/func_80A4EE48.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mkk/func_80A4EEF4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mkk/func_80A4EF74.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mkk/func_80A4F16C.s")
+static struct EnMkkDlists sBoeDLists[] = {
+    { object_mkk_DL_000030, object_mkk_DL_0000B0, object_mkk_DL_0000C8, object_mkk_DL_000140 },
+    { object_mkk_DL_0001F0, object_mkk_DL_000278, object_mkk_DL_000290, object_mkk_DL_000310 },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mkk/EnMkk_Draw.s")
+static Color_RGBA8 D_80A4F7C4[] = {
+    { 255, 255, 255, 255 }, { 128, 128, 128, 255 }, { 0, 0, 0, 255 },
+    { 0, 0, 0, 255 },       { 128, 128, 128, 255 }, { 255, 255, 255, 255 },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mkk/func_80A4F4C8.s")
+void EnMkk_Init(Actor* thisx, PlayState* play) {
+    EnMkk* this = THIS;
+    s32 params8;
+    s32 params2;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Collider_InitAndSetSphere(play, &this->collider, &this->actor, &sSphereInit);
+    this->collider.dim.worldSphere.radius = sSphereInit.dim.modelSphere.radius;
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    ActorShape_Init(&this->actor.shape, 1000.0f, NULL, 0.0f);
+
+    this->primColorSelect = 0;
+    this->unk_149 = 0;
+    this->unk_152 = (this->actor.shape.rot.x >= 0) ? this->actor.shape.rot.x : this->actor.shape.rot.x * -0x50;
+
+    this->unk_14C = CLAMP(this->actor.shape.rot.z, 0, 16);
+
+    this->actor.shape.rot.z = 0;
+    this->actor.world.rot.z = 0;
+    this->actor.shape.rot.x = 0;
+    this->actor.world.rot.x = 0;
+
+    Math_Vec3f_Copy(&this->unk_154, &this->actor.world.pos);
+    Math_Vec3f_Copy(&this->unk_160, &this->actor.world.pos);
+    this->unk_14B = (this->actor.params & 4) ? 8 : 0;
+
+    params8 = (this->actor.params >> 8) & 0xFF;
+    params2 = this->actor.params & 2;
+    this->actor.params &= 1;
+    if (this->actor.params == 1) {
+        this->actor.hintId = 0x3C;
+    } else {
+        this->actor.hintId = 0x2C;
+    }
+    if ((params8 == 0) || (params8 == 255)) {
+        func_80A4E0CC(this);
+        this->unk_178 = 30000.0f;
+        if (params2 > 0) {
+            this->unk_14B |= 4;
+        }
+    } else {
+        this->unk_178 = params8 * 40.0f * 0.1f;
+        func_80A4EDF0(this);
+        this->unk_14E = 0;
+    }
+}
+
+void EnMkk_Destroy(Actor* thisx, PlayState* play) {
+    EnMkk* this = THIS;
+
+    Collider_DestroySphere(play, &this->collider);
+}
+
+void func_80A4E0CC(EnMkk* this) {
+    this->alpha = 0;
+    this->unk_14B |= 3;
+    this->actor.flags &= ~ACTOR_FLAG_1;
+    this->actionFunc = func_80A4E100;
+}
+
+void func_80A4E100(EnMkk* this, PlayState* play) {
+    s32 newAlpha;
+
+    if ((this->unk_14B & 4) != 0) {
```

Comment:
```suggestion
    if (this->unk_14B & 4) {
```
This looks like a flag, so you don't need the `!= 0`